### PR TITLE
Feature Request: Buttons & Button Set

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, the button styles and design are littered over the foundation-demo codebase. Let's refactor to a singular button design.

Issue Number: #59 

## What is the new behavior?
A singlular daff-button component that can be applied to a button, a an element with a aria-role="button" that applies consistent styling, animations, and interactions.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/design/atoms/atoms.module.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@NgModule({
+    imports: [
+      CommonModule,
+      FormsModule
+    ],
+    declarations: [
+      ButtonComponent
+    ],
+    exports: [
+      ButtonComponent
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/design/atoms/atoms.module.ts]: The error is coming from here. You're declaring it here and in the button.module. You should import `ButtonModule` here instead.

File: apps/foundation-demo/src/app/design/atoms/atoms.module.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+      ButtonModule
     ],
-    declarations: [
-      ButtonComponent
+    declarations: [ 
+
     ],
     exports: [
-      ButtonComponent
+
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/design/atoms/atoms.module.ts]: This needs to export ButtonModule

File: apps/foundation-demo/src/app/design/atoms/button/button.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+];
+
+
+@Component({
+  selector: 'button[daff-button], button[daff-stroked-button], button[daff-clear-button], button[daff-icon-button]' + ',' +
+            'a[daff-button], a[daff-stroked-button], a[daff-clear-button], a[daff-icon-button]',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/design/atoms/button/button.component.ts]: So you have the styles for a class called `daff-button`, but those styles won't be applied unless you manually add the class:
```
<button daff-button class="**daff-button**">Add to Cart</button>
```
instead, you could add the class to the host button upon creation of a new `daff-button` component. To do this, you add `host: {'class': 'daff-button'}` here.

File: apps/foundation-demo/src/app/design/atoms/atoms.module.ts (hunk 1 above)
Comment (damienwebdev) [reply apps/foundation-demo/src/app/design/atoms/atoms.module.ts]: Let's remove this module entirely. Nothing should be importing this whole thing. Components should import as needed to reduce bundle size.

**Comment by xelaint** [comment]
@damienwebdev want to add an `expanded` variant.

**Comment by damienwebdev** [comment]
@xelaint What's still outstanding on this PR? Just color support?

**Comment by xelaint** [comment]
@damienwebdev @lderrickable ready for review

**Comment by lderrickable** [comment]
:(

**Pull request merged**